### PR TITLE
fix(bench): don't throw on missing ops

### PR DESCRIPTION
**Pull request by danteissaias**

### CODE REVIEW

**Comment by danteissaias** [comment]
Fixes https://github.com/denoland/deno/issues/14862

File: runtime/js/40_testing.js [hunk 1]
```diff
@@ -749,15 +748,19 @@
...
-    benchDesc.filteredOut = filteredOut;
+      const { id, filteredOut } = core.opSync("op_register_bench", benchDesc);
+      benchDesc.id = id;
+      benchDesc.filteredOut = filteredOut;
 
-    ArrayPrototypePush(benchDescs, benchDesc);
+      ArrayPrototypePush(benchDescs, benchDesc);
+    } catch {
+      // Ignore missing ops
+    }
```
Comment (bartlomieju) [inline runtime/js/40_testing.js:763]: This is brittle, it will ignore all errors that could happen during registration

**Review by bartlomieju — COMMENTED** [review]
@danteissaias thank you for the PR, however I think this approach is not correct. I did open https://github.com/denoland/deno/pull/14946 with alternative approach that I think is simpler and more bullet proof

**Comment by danteissaias** [comment]
> @danteissaias thank you for the PR, however I think this approach is not correct. I did open https://github.com/denoland/deno/pull/14946 with alternative approach that I think is simpler and more bullet proof

@bartlomieju I agree your approach is better however bench should be moved to the unstable ns and `op_bench_check_unstable` should be removed. Should I open a new PR with only these changes?

**Comment by bartlomieju** [comment]
> > @danteissaias thank you for the PR, however I think this approach is not correct. I did open #14946 with alternative approach that I think is simpler and more bullet proof
> 
> @bartlomieju I agree your approach is better however bench should be moved to the unstable ns and `op_bench_check_unstable` should be removed. Should I open a new PR with only these changes?

I think that's not necessary - if we do that, then you will get errors saying that `Deno.bench` is undefined (unless you specify `--unstable` flag). I believe it's better to print error and exit when first bench is run. And for cases where you have `Deno.bench()` registered but use `deno run` would still work as expected. WDYT?

**Comment by danteissaias** [comment]
> > > @danteissaias thank you for the PR, however I think this approach is not correct. I did open #14946 with alternative approach that I think is simpler and more bullet proof
> > 
> > 
> > @bartlomieju I agree your approach is better however bench should be moved to the unstable ns and `op_bench_check_unstable` should be removed. Should I open a new PR with only these changes?
> 
> I think that's not necessary - if we do that, then you will get errors saying that `Deno.bench` is undefined (unless you specify `--unstable` flag). I believe it's better to print error and exit when first bench is run. And for cases where you have `Deno.bench()` registered but use `deno run` would still work as expected. WDYT?

This isn’t consistent with other unstable APIs which are undefined without `--unstable`. I guess we could treat it as a special case but I don’t see much benefit.

**Comment by bartlomieju** [comment]
> This isn’t consistent with other unstable APIs which are undefined without --unstable. I guess we could treat it as a special case but I don’t see much benefit.

That's a valid point, but I think it's fine band-aid solution, since we would like to stabilize `deno bench` soon anyway.

**Comment by danteissaias** [comment]
> > This isn’t consistent with other unstable APIs which are undefined without --unstable. I guess we could treat it as a special case but I don’t see much benefit.
> 
> That's a valid point, but I think it's fine band-aid solution, since we would like to stabilize `deno bench` soon anyway.

Fair

**Pull request closed without merging**